Libraries/Microsoft.Teams.Apps/AppPlugins.cs

```diff
@@ -26,36 +27,34 @@ public partial class App
 
     public App AddPlugin(IPlugin plugin)
     {
-        var attr = GetPluginAttribute(plugin);
+        var attr = PluginService.GetAttribute(plugin);
 
         // broadcast plugin events
-        plugin.Events += Events.Emit;
+        plugin.Events += async (plugin, name, @event, token) =>
+        {
+            var eventType = new EventType(name);
+
+            await Events.Emit(plugin, $"{attr.Name}.{name}", @event, token);
+
+            if (eventType.IsBuiltIn && !eventType.IsStart)
+            {
+                return await Events.Emit(plugin, name, @event, token);
+            }
```

Comment:
So if i have an event "Activity" in my plugin, how does the handler know what the payload will be for "Activity"?

Reply:
each route has a way of casting the base activity context to its more specific activity type 
https://github.com/microsoft/teams.net/blob/main/Libraries/Microsoft.Teams.Apps/Routing/Route.cs#L56
https://github.com/microsoft/teams.net/blob/main/Libraries/Microsoft.Teams.Apps/Routing/Activities/CommandActivity.cs#L8